docs/architecture/dapr-for-net-developers/publish-subscribe-buildingblock.md

```diff
@@ -0,0 +1,398 @@
+
+``` http
+http://localhost:<appPort>/dapr/subscribe
+```
+
+- `<appPort>` informs the Dapr sidecar of the port upon which the application is listening.
+
+You can implement this endpoint yourself. But Dapr provides more intuitive ways of implementing it. We'll address this functionality later in this chapter.
+
+The response from the call contains a list of topics to which the applications will subscribe. Each includes an endpoint to call when the topic receives a message. Here's an example of a response:
+
+```json
+[
+  {
+    "pubsubname": "pubsub",
+    "topic": "newOrder",
+    "route": "/orders"
+  },
+  {
+    "pubsubname": "pubsub",
+    "topic": "newProduct",
+    "route": "/productCatalog/products"
+  }
+]
+```
+
+In the JSON response, you can see the application wants to subscribe to topics `newOrder` and `newProduct`. It registers the endpoints `/orders` and `/productCatalog/products` for each, respectively. For both subscriptions, the application is binding to the Dapr component named `pubsub`.
+
+Figure 7-3 presents the flow of the example.
+
+![Example Publish/Subscribe flow with Dapr](media/pubsub-dapr-pattern.png)
+
+**Figure 7-3**. Publish/subscribe flow with Dapr.
+
+From the previous figure, note the flow:
+
+1. The Dapr sidecar for Service B calls the `/dapr/subscribe` endpoint from Service B (the consumer). The service responds with the subscriptions it wants to create.
+2. The Dapr sidecar for Service B creates the requested subscriptions on the message broker.
+3. Service A publishes a message at the `/v1.0/publish/<pubsubname>/<topic>` endpoint on the Dapr Service A sidecar.
+4. The Service A sidecar publishes the message to the message broker.
+5. The message broker sends a copy of the message to the Service B sidecar.
+6. The Service B sidecar calls the endpoint corresponding to the subscription (in this case `/orders`) on Service B.
+
+Making HTTP calls against the native Dapr APIs is time-consuming and abstract. Your calls are crafted at the HTTP level, and you'll need to handle plumbing concerns such as serialization and HTTP response codes. Fortunately, there's a more intuitive way. Dapr provides several language-specific SDKs for popular development platforms. At the time of this writing, Go, Node.js, Python, .NET, Java, and JavaScript are available.
+
+### Using the Dapr .NET SDK
+
+For .NET Developers, the Dapr .NET SDK provides a more productive way of working with Dapr. The SDK exposes a `DaprClient` class through which you can directly invoke Dapr functionality. It's intuitive and easy to use.
+
+To publish a message, the `DaprClient` exposes a `PublishEventAsync` method. It's straightforward to use:
+
+```csharp
+var data = new OrderData
+{
+  id = "123456",
+  productId = "67890",
+  amount = 2
+}
+
+await PublishEventAsync<OrderData>("pubsub", "newOrder", data);
```

Comment:
Don't you need a `DaprClient` instance to call the `PublishEventAsync` method on?

Reply:
Yes, indeed! I will update this snippet.